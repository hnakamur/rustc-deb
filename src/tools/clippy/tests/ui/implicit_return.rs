#![warn(clippy::implicit_return)]

fn test_end_of_fn() -> bool {
    if true {
        // no error!
        return true;
    }
    true
}

#[allow(clippy::needless_bool)]
fn test_if_block() -> bool {
    if true {
        true
    } else {
        false
    }
}

#[allow(clippy::match_bool)]
#[rustfmt::skip]
fn test_match(x: bool) -> bool {
    match x {
        true => false,
        false => { true },
    }
}

#[allow(clippy::never_loop)]
fn test_loop() -> bool {
    loop {
        break true;
    }
}

#[allow(clippy::never_loop)]
fn test_loop_with_block() -> bool {
    loop {
        {
            break true;
        }
    }
}

#[allow(clippy::never_loop)]
fn test_loop_with_nests() -> bool {
    loop {
        if true {
            break true;
        } else {
            let _ = true;
        }
    }
}

fn test_closure() {
    #[rustfmt::skip]
    let _ = || { true };
    let _ = || true;
}

fn main() {
    let _ = test_end_of_fn();
    let _ = test_if_block();
    let _ = test_match(true);
    let _ = test_loop();
    let _ = test_loop_with_block();
    let _ = test_loop_with_nests();
    test_closure();
}
