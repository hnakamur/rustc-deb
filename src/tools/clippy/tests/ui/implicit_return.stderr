error: missing return statement
  --> $DIR/implicit_return.rs:8:5
   |
LL |     true
   |     ^^^^ help: add `return` as shown: `return true`
   |
   = note: `-D clippy::implicit-return` implied by `-D warnings`

error: missing return statement
  --> $DIR/implicit_return.rs:14:9
   |
LL |         true
   |         ^^^^ help: add `return` as shown: `return true`

error: missing return statement
  --> $DIR/implicit_return.rs:16:9
   |
LL |         false
   |         ^^^^^ help: add `return` as shown: `return false`

error: missing return statement
  --> $DIR/implicit_return.rs:24:17
   |
LL |         true => false,
   |                 ^^^^^ help: add `return` as shown: `return false`

error: missing return statement
  --> $DIR/implicit_return.rs:25:20
   |
LL |         false => { true },
   |                    ^^^^ help: add `return` as shown: `return true`

error: missing return statement
  --> $DIR/implicit_return.rs:32:9
   |
LL |         break true;
   |         ^^^^^^^^^^ help: change `break` to `return` as shown: `return true`

error: missing return statement
  --> $DIR/implicit_return.rs:40:13
   |
LL |             break true;
   |             ^^^^^^^^^^ help: change `break` to `return` as shown: `return true`

error: missing return statement
  --> $DIR/implicit_return.rs:49:13
   |
LL |             break true;
   |             ^^^^^^^^^^ help: change `break` to `return` as shown: `return true`

error: missing return statement
  --> $DIR/implicit_return.rs:58:18
   |
LL |     let _ = || { true };
   |                  ^^^^ help: add `return` as shown: `return true`

error: missing return statement
  --> $DIR/implicit_return.rs:59:16
   |
LL |     let _ = || true;
   |                ^^^^ help: add `return` as shown: `return true`

error: aborting due to 10 previous errors

